test/environment_variables.cc

```diff
@@ -0,0 +1,118 @@
+#include <array>
+
+#include "gtest/gtest.h"
+#include "stout/flags.h"
+#include "test/test.pb.h"
+
+// On Windows there is no `setenv()` or `unsetenv()`.
+// Instead we use `_putenv_s()` function.
+#ifdef _WIN32
+int setenv(
+    const char* env_name,
+    const char* env_value,
+    int replace) {
+  return _putenv_s(env_name, env_value);
+}
+
+int unsetenv(const char* env_name) {
+  return _putenv_s(env_name, "");
+}
+#endif
+
+TEST(FlagsTest, EnvironmentVariableString) {
+  test::Flags flags;
+
+  auto parser = stout::flags::Parser::Builder(&flags)
+                    .IncludeEnvironmentVariablesWithPrefix("STOUT_FLAGS_TEST")
+                    .Build();
+
+  std::array arguments = {
+      "program",
+      "--bar",
+      "one",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  const char* env_name = "STOUT_FLAGS_TEST_FOO";
+  const char* env_value = "'HELLO world'";
+  int replace = 0;
+
+  setenv(env_name, env_value, replace);
+  parser.Parse(&argc, &argv);
+  unsetenv(env_name);
+
+  EXPECT_TRUE(flags.bar());
+  EXPECT_EQ(2, argc);
+  EXPECT_EQ("HELLO world", flags.foo());
+  EXPECT_EQ("program", argv[0]);
+  EXPECT_EQ("one", argv[1]);
+}
+
+TEST(FlagsTest, EnvironmentVariableStringFailure) {
+  test::Flags flags;
+
+  auto parser = stout::flags::Parser::Builder(&flags)
+                    .IncludeEnvironmentVariablesWithPrefix(
+                        "STOUT_FLAGS_TEST")
+                    .Build();
+
+  std::array arguments = {
+      "program",
+      "one",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  const std::string regex =
+      "program: Failed while parsing and validating flags:"
+      "\?\n\?\n"
+      ". Failed to parse flag 'foo' from normalized value 'HELLO' "
+      "due to protobuf text-format parser error.s.: "
+      "Expected string, got: HELLO"
+      "\?\n\?\n"
+      ". Flag 'foo' not parsed but required";
+
+  const char* env_name = "STOUT_FLAGS_TEST_FOO";
+  const char* env_value = "HELLO";
+  int replace = 0;
+
+  setenv(env_name, env_value, replace);
+
+  EXPECT_DEATH(
+      parser.Parse(&argc, &argv),
+      testing::ContainsRegex(regex));
+
+  unsetenv(env_name);
+}
+
+TEST(FlagsTest, EnvironmentVariableWithNoUnderlineInName) {
```

Comment:
Good catch! Let's actually not support this case, let's require there to be an underscore between the prefix and the variable name.